### PR TITLE
more 0.13 docs updates (FAQ and k8s-plugins)

### PR DESCRIPTION
**Pull request by Orzelius**

### CODE REVIEW

File: docs/misc/faq.md [hunk 1]
```diff
@@ -7,69 +7,57 @@ title: FAQ
```
Comment (Orzelius) [inline docs/misc/faq.md:7]: I actually think most of this FAQ doc should be dissolved into the related documentation sections.

If a user has a question about some garden functionality they'll first look at the documentation about that functionality and that's where the answer should be.

File: docs/misc/faq.md [hunk 2]
```diff
@@ -151,63 +144,60 @@ See [this section](../k8s-plugins/advanced/in-cluster-building.md#pulling-base-i
...
-### If tests have a task as a dependency, is the task re-run every time before the test?
-
-It is, which is why we recommend that tasks are written to be idempotent. Tasks by nature don’t really have a status check, unlike services.
+So what we generally do is to make sure Runs are idempotent and exit early if they shouldn't run again. But that means the process still needs to be started, which is of course slower than not doing it at all.
 
-### Why is a task not triggered on changes in watch mode?
+### If a Test has a Run as a dependency, is the Run re-run every time before the Test?
 
-This is intentional, we don't re-run tasks on file watch events. We debated this behavior quite a bit and ultimately opted not to run task dependencies on every watch event.
+It is, which is why we recommend that tasks are written to be idempotent. Tasks by nature don’t really have a status check, unlike Deploys.
```
Comment (Walther) [inline docs/misc/faq.md:157]: nit: should we call these Runs or Run tasks? (i vaguely recall "tasks" being a deprecated term, but i could misremember here)

**Review by Walther — APPROVED** [review]
Looks good, thanks for the update! ✨ 

One non-blocking nit, could even be ignored.

**Review by vvagaytsev — DISMISSED** [review]
Great job, thank you! 🚀

File: docs/misc/faq.md (hunk 2 above)
Comment (vvagaytsev) [reply docs/misc/faq.md:157]: I think it's better to use "Runs" here. By the way, there are some failures in the [check-docs](https://app.circleci.com/pipelines/github/garden-io/garden/17449/workflows/edf2fd57-c9ad-49b3-b5ac-6bc315ac9fd4/jobs/301713), I'll take a look.

Comment (vvagaytsev) [reply docs/misc/faq.md:157]: Updated the TOC in https://github.com/garden-io/garden/pull/4307/commits/498c90af670305c66ec99dc5cf2736d0949b517a

Comment (Orzelius) [reply docs/misc/faq.md]: [a1ce7e4](https://github.com/garden-io/garden/pull/4307/commits/a1ce7e49448cfbb44fb9a0ede19a69739317d0bd)

Fixed the terminology

File: docs/misc/faq.md (hunk 1 above)
Comment (Orzelius) [reply docs/misc/faq.md:7]: I will do this in a separate pr

**Pull request merged**